src/main/java/serverutils/pregenerator/ChunkLoaderManager.java

```diff
@@ -0,0 +1,186 @@
+package serverutils.pregenerator;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.Vector;
+
+import net.minecraft.server.MinecraftServer;
+
+import org.apache.commons.lang3.tuple.Pair;
+
+import serverutils.lib.util.misc.PregeneratorCommandInfo;
+import serverutils.pregenerator.filemanager.PregeneratorFileManager;
+
+public class ChunkLoaderManager {
+
+    public final static ChunkLoaderManager instance = new ChunkLoaderManager();
+    private boolean isGenerating = false;
+    private int dimensionID;
+    private MinecraftServer serverType;
+    private Vector<Pair<Integer, Integer>> chunksToLoad = new Vector<>(1000);
```

Comment:
Using `IntIntPair` from Fastutils is better here to avoid unnecessary boxing

Reply:
+1 to packing the coords into a long.  See https://github.com/GTNewHorizons/GTNHLib/blob/master/src/main/java/com/gtnewhorizon/gtnhlib/util/CoordinatePacker.java

Reply:
Is there a nicer way to track the chunks to load? I mean I could use an array/list, but I thought the dynamic sizing of the vector would be nice. Not a java expert, so I don't know what is appropriate to use here.

Reply:
An ArrayList is perfectly fine here, a LongArrayList from fastutil is the most efficient in this case.

Reply:
A Vector is basically a slower ArrayList, because it does full thread synchronization on every operation

Reply:
> A Vector is basically a slower ArrayList, because it does full thread synchronization on every operation

> +1 to packing the coords into a long. See https://github.com/GTNewHorizons/GTNHLib/blob/master/src/main/java/com/gtnewhorizon/gtnhlib/util/CoordinatePacker.java

Very cool stuff. Love the coordinates as a long. Also didn't know this about Vector in java. C++ doesn't have fancy thread synchronization for you :). Thanks for the info!